components/org.wso2.carbon.identity.application.authenticator.adapter/src/main/java/org/wso2/carbon/identity/application/authenticator/adapter/util/AuthenticatedUserBuilder.java

```diff
@@ -182,6 +187,13 @@ private Map<ClaimMapping, String> resolveUserNameAndClaimsFromResponse() {
         return userAttributes;
     }
 
+    private void resolveGroupsForFederatedUser(Map<ClaimMapping, String> claimMappings) {
+
+        if (userFromResponse.getUser().getGroups() != null) {
+            claimMappings.put(buildClaimMapping(GROUP_CLAIM), String.join(",", userFromResponse.getUser().getGroups()));
```

Comment:
Check if "," is the default separator used. Or whether it's a configurable param like multi attribute separator.
And I hope "," is not allowed in groups and if so this validation should happen. Else in framework it will break